control-plane/pkg/prober/async_prober.go

```diff
@@ -161,3 +165,28 @@ func (a *asyncProber) Probe(ctx context.Context, addressable Addressable, expect
 func (a *asyncProber) enqueueArg(_ string, arg interface{}) {
 	a.enqueue(arg.(types.NamespacedName))
 }
+
+// Resets the cache to a new empty cache with the given expiry time
+func (a *asyncProber) resetCache(ctx context.Context, expiry time.Duration) {
+	a.clientMu.Lock()
+	defer a.clientMu.Unlock()
+	a.cache = NewLocalExpiringCache(ctx, expiry)
+}
```

Comment:
Is this method used only for testing?

Reply:
Yeah this method is currently only used for tests to ensure that the prober re-probes the pod with the new certificates instead of falling back to the cached state. It could be useful if we want to always reset the prober cache when the CA certs get rotated, to ensure that everything is still reachable. WDYT?

Reply:
Did you consider setting a relatively low expiry time at the creation time and wait for it to expire for tests?

Making the code more testable is absolutely necessary, what I'd avoid is creating a different code path that is just used for testing